src/mono/mono/mini/interp/transform.c

```diff
@@ -4429,6 +4480,18 @@ generate_code (TransformData *td, MonoMethod *method, MonoMethodHeader *header,
 			continue;
 		}
 
+		// Skip blocks with uninitialized stack
+		if (td->cbb->stack_height == -1) {
+			int bb_size = bb->end-bb->start;
+			g_assert (bb_size > 0);
+			td->ip += bb_size;
+
+			skipped_bbs_present = TRUE;
+			link_bblocks = FALSE;
+			emitting = FALSE;
+			continue;
```

Comment:
The loop for generate_code is depending on a lot of local state in all of these boolean flags.

It would be good to do two things:
1. add a comment at the top of the loop with all the flag variables that we expect to be reset to different values.
2. Think about if we can have a state enum that is computed just from the basic block structure and the previous state, and compute it once at the beginning of the loop so that we don't have this "assign to 2 or 3 flag booleans and goto the top of the loop" pattern which is fragile, hard to debug, and depends on the execution of a bunch of code to decide what to do next.
(don't need to do it in this PR, but maybe something for the future).